--- rosboost_cfg.py	2011-01-25 23:09:57.241281018 +0900
+++ ../rosboost_cfg.py	2011-01-25 23:11:05.744405515 +0900
@@ -177,13 +177,13 @@
     return result[-1]
 
 def search_paths(sysroot):
-    _search_paths = [(None if 'ROS_BOOST_ROOT' not in os.environ else os.environ['ROS_BOOST_ROOT'], False),
-                 (None if 'INCLUDE_DIRS' not in os.environ else os.environ['INCLUDE_DIRS'], True), 
-                 (None if 'CPATH' not in os.environ else os.environ['CPATH'], True),
-                 (None if 'C_INCLUDE_PATH' not in os.environ else os.environ['C_INCLUDE_PATH'], True),
-                 (None if 'CPLUS_INCLUDE_PATH' not in os.environ else os.environ['CPLUS_INCLUDE_PATH'], True),
-                 (sysroot+'/usr', True), 
-                 (sysroot+'/usr/local', True)]
+    _search_paths = [(None if 'INCLUDE_DIRS' not in os.environ else os.environ['INCLUDE_DIRS'], True), 
+             (None if 'CPATH' not in os.environ else os.environ['CPATH'], True),
+             (None if 'C_INCLUDE_PATH' not in os.environ else os.environ['C_INCLUDE_PATH'], True),
+             (None if 'CPLUS_INCLUDE_PATH' not in os.environ else os.environ['CPLUS_INCLUDE_PATH'], True),
+             (sysroot+'/usr', True), 
+             (sysroot+'/usr/local', True),
+             (None if 'ROS_BOOST_ROOT' not in os.environ else os.environ['ROS_BOOST_ROOT'], False)]
 
     search_paths = []
     for (str, system) in _search_paths:
@@ -209,14 +209,14 @@
     if (ver.is_system_install):
         dynamic_search_paths = ["libboost_%s-mt.%s"%(name, lib_suffix),
                                 "libboost_%s.%s"%(name, lib_suffix)]
-        static_search_paths = ["libboost_%s-mt.a"%(name),
+        static_search_paths = ["libboost_%s*-mt.a"%(name),
                                "libboost_%s.a"%(name)]
     else:
         dynamic_search_paths = ["libboost_%s*%s_%s*.%s"%(name, ver.major, ver.minor, lib_suffix),
                                 "libboost_%s-mt*.%s"%(name, lib_suffix),
                                 "libboost_%s*.%s"%(name, lib_suffix)]
         static_search_paths = ["libboost_%s*%s_%s*.a"%(name, ver.major, ver.minor),
-                               "libboost_%s-mt*.a"%(name),
+                               "libboost_%s*-mt*.a"%(name),
                                "libboost_%s*.a"%(name)]
         
     # Boost.Python needs some special handling on some systems (Karmic), since it may have per-python-version libs
@@ -323,7 +323,6 @@
         return
        
     ver = find_boost(search_paths(options.sysroot))
-    
     if (ver is None):
         raise BoostError("Cannot find boost in any of %s"%search_paths(options.sysroot))
         sys.exit(0)
